### ISSUE TITLE
Mapping matrix broken after upgrade to version 3.17.*

### ISSUE THREAD

**Issue by bednic** [doctrine/doctrine2#11106]
### BC Break Report

|    Q        |   A
|------------ | ------
| BC Break    | yes
| Version     | 3.17.*

#### Summary

As described in [documentation ](https://www.doctrine-project.org/projects/doctrine-dbal/en/3.7/reference/types.html#mapping-matrix) PHP data type `mixed` should be possible map to `JSON` database type. After upgrading to 3.17.1 version this is not true anymore, and **orm:schema-validate** writes error because database json can now be mapped **only** to PHP array data type. Thus I lost posibility to use PHP mixed data type.

#### Previous behavior

Map PHP mixed data type to database JSON type.

#### Current behavior

ORM schema validation shows errors, because database JSON type can be mapped only to PHP array data type.

#### How to reproduce
```php
#[ORM\Entity(SystemPropertyRepository::class)]
#[ORM\Table(schema: 'dw')]
class SystemProperty extends DataEntityWithKey
{
    /**
     * @var array value
     */
    #[ORM\Column(type: 'json')] //<-- this should be possible
    public mixed $value;
}

```

**Comment by greg0ire** [comment]
Is this fixed by #11076?

**Comment by bednic** [comment]
Yes, thanks! I'm closing this issue then.
